tests/sentry/api/endpoints/test_project_stacktrace_link.py

```diff
@@ -213,3 +214,163 @@ def test_file_no_stack_root_match(self, mock_integration):
+        self.repo.provider = "example"
+        self.repo.save()
+
+        self.code_mapping1 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="",
+            source_root="",
+        )
+
+        self.login_as(self.user)
+
+    def expected_configurations(self, code_mapping) -> Mapping[str, Any]:
+        return {
+            "defaultBranch": "master",
+            "id": str(code_mapping.id),
+            "integrationId": str(self.integration.id),
+            "projectId": str(self.project.id),
+            "projectSlug": self.project.slug,
+            "provider": serialized_provider(),
+            "repoId": str(self.repo.id),
+            "repoName": self.repo.name,
+            "sourceRoot": code_mapping.source_root,
+            "stackRoot": code_mapping.stack_root,
+        }
+
+    def test_file_not_found_munge_frame_fallback_success_android(self):
+        ctx = {
+            "file": "file.java",
+            "filename": "file.java",
+            "commit_id": None,
+            "platform": "java",
+            "sdk_name": None,
+            "abs_path": None,
+            "module": "usr.src.getsentry.file",
+            "package": None,
+        }
+
+        munged_url = try_path_munging(self.code_mapping1, "file.java", ctx)
+
+        with mock.patch.object(
+            ExampleIntegration,
+            "get_stacktrace_link",
+            return_value="https://example.com/getsentry/sentry/blob/master/usr/src/getsentry/file.java",
+        ):
+            response = self.get_success_response(
+                self.organization.slug,
+                self.project.slug,
+                qs_params={
+                    "file": "file.java",
+                    "module": "usr.src.getsentry.file",
+                    "platform": "java",
+                },
+            )
+            assert response.data["config"] == self.expected_configurations(self.code_mapping1)
+            assert (
+                response.data["sourceUrl"]
+                == "https://example.com/getsentry/sentry/blob/master/usr/src/getsentry/file.java"
+            )
+            assert response.data["integrations"] == [serialized_integration(self.integration)]
+            assert (
+                munged_url["attemptedUrl"]
+                == "https://example.com/getsentry/sentry/blob/master/usr/src/getsentry/file.java"
+            )
+
+    def test_file_not_found_munge_frame_fallback_success_cocoa(self):
+        ctx = {
+            "file": "AppDelegate.swift",
+            "filename": "AppDelegate.swift",
+            "commit_id": None,
+            "platform": "cocoa",
+            "sdk_name": None,
+            "abs_path": "/Users/gszeto/code/SwiftySampleProject/SampleProject/Classes/App Delegate/AppDelegate.swift",
+            "module": None,
+            "package": "SampleProject",
+        }
```

Comment:
Similar comment as what I said on the first test. The same for all following tests.

Reply:
Thank you for your helpful feedback! I have implemented the changes we've discussed